api/src/main/java/org/apache/iceberg/actions/ActionsProvider.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions;
+
+import org.apache.iceberg.Table;
+
+/**
+ * An API that should be implemented by query engine integrations for providing actions.
+ */
+public interface ActionsProvider {
+
+  /**
+   * Instantiates an action to snapshot an existing table.
+   */
+  default SnapshotTable snapshotTable(String sourceTableIdent, String destTableIdent) {
+    throw new UnsupportedOperationException(this.getClass().getName() + " does not implement snapshotTable");
+  }
+
+  /**
+   * Instantiates an action to migrate an existing table to Iceberg.
+   */
+  default MigrateTable migrateTable(String tableIdent) {
```

Comment:
I don't really like using Strings as identifiers as it requires implementations to parse it but I am not sure other options are better. I considered using `TableIdentifier` Iceberg defines but it is in the catalog module so it will be a bit weird to use it here. Also, parsing seems inevitable as there may be engine-specific parsing.

Reply:
Yeah, I think a String is a reasonable choice here.

Reply:
Ok, let me resolve this one then.